post-install-setup.sh

```diff
@@ -0,0 +1,237 @@
+#		- Check the root user
+#		- Check the liota user
+#		- Check the liota group
+#	- Configuration files: /etc/liota
+#		- If /etc/liota doesn't exist, make it
+#		- Change the owner and group to what we are expecting
+#		- Change the permissions on the configuration directory
+#		- Check if the example config exists
+#		- Copy the exapmle to the config dir
+#		- Change the owner and group to what we are expecting
+#		- Change the permissions on the config file
+#
+# ----------------------------------------------------------------------------
+#
+# Defined assumptions follow, these are all overrideable with environment 
+# variables
+#
+
+# Defines where the configuration directory should be
+#	Default: /etc/liota
+_LIOTA_ETCDIR=${LIOTA_ETCDIR:-/etc/liota}
+
+# Defines who the liota user is, so that appropriate
+# permissions can be set
+#	Default: liota
+_LIOTA_USER=${LIOTA_USER:-liota}
+
+# Defines who the 'root' user is
+# It's assumed that things like /etc/liota will be owned by root but readable
+# by the user.  
+#	Default: root
+_LIOTA_ROOTUSER=${LIOTA_ROOTUSER:-root}
+
+# Defines that the permissions for the etc directory
+#	Default: 0750
+_LIOTA_ETCPERMS=${LIOTA_ETCPERMS:-0750}
+
+# Defines where the example config file is at
+#	Default:
+_LIOTA_CONFFROM=${LIOTA_CONFFROM:-/usr/lib/liota/config/liota.conf}
+
+# Defines where the config file should live
+#	Default: /etc/liota/liota.conf
+_LIOTA_CONFTO=${LIOTA_CONFTO:-/etc/liota/liota.conf}
+
+# Defines what the permissions for the config file should be
+#	Default: 0750
+_LIOTA_CONFPERMS=${LIOTA_CONFPERMS:-0750}
+
+echo "Liota: etc dir: ${_LIOTA_ETCDIR}"
+
+# ----------------------------------------------------------------------------
+#
+# No User Serviceable Parts Beyond This Point
+#	Qualified service personal only
+#
+# ----------------------------------------------------------------------------
+
+#
+# Check if the users in question all actually exist
+#
+
+which getent &> /dev/null
+
+if [[ "$?" != "0" ]]
+then
+	echo "command \`which\` is not installed or available in the path. Can't perform checks. This is a fatal error."
+	exit
+fi
+
+getent passwd "${_LIOTA_ROOTUSER}" &> /dev/null
+
+if [[ "$?" != "0" ]]
+then
+	echo "'Root' User ${_LIOTA_ROOTUSER} is not setup, this is odd and needs to be resolved. This is a fatal error."
+	exit
+fi
+
+getent passwd "${_LIOTA_USER}" &> /dev/null
+
+if [[ "$?" != "0" ]]
+then
+	echo "Liota User ${_LIOTA_USER} is not setup, this is odd and needs to be resolved. This is a fatal error."
+	exit
+fi
+
+getent group "${_LIOTA_USER}" &> /dev/null
+
+if [[ "$?" != "0" ]]
+then
+	echo "Liota Group ${_LIOTA_USER} is not setup, this is odd and needs to be resolved. This is a fatal error."
+	exit
+fi
+
+
+#
+# Configuration directory
+#
+
+if [[ -e "${_LIOTA_ETCDIR}" ]]
+then
+	echo "ERROR: ${_LIOTA_ETCDIR} already exists, and assuming things are setup already or not but automatic running will not proceed. This is a fatal error."
```

Comment:
This is not a production deployment script, it's a dev environment helper. Even in a production deployment the package manager world *NEVER* replace the conf file with a new one. Distros like Debian can try to patch in new config info, but that has mixed results and usually requires manual intervention anyway.

If there's enough churn in the config file it basically will require manual user intervention to merge the respective changes, so no I'm not going to replace the config file if one exists, and a user would have to copy it in themselves